src/bot.ts

```diff
@@ -1177,6 +1192,146 @@ export class DiscordBot {
         await this.OnMessage(newMsg);
     }
 
+    public async OnMessageReactionAdd(reaction: Discord.MessageReaction, user: Discord.User | Discord.PartialUser) {
+        const message = reaction.message;
+        const reactionName = reaction.emoji.name;
+        log.verbose(`Got message reaction add event for ${message.id} with ${reactionName}`);
+
+        let rooms: string[];
+
+        try {
+            rooms = await this.channelSync.GetRoomIdsFromChannel(message.channel);
+        } catch (err) {
+            log.verbose(`No bridged rooms to forward reaction to. Reaction Event: ${reaction}`);
+            MetricPeg.get.requestOutcome(message.id, true, "dropped");
+            return;
+        }
+
+        const intent = this.GetIntentFromDiscordMember(user);
+        await intent.ensureRegistered();
+        this.userActivity.updateUserActivity(intent.userId);
+
+        const storeEvent = await this.store.Get(DbEvent, {
+            discord_id: message.id
+        });
+
+        if (!storeEvent?.Result) {
+            log.verbose(`Received add reaction event for untracked message. Dropping! Reaction Event: ${reaction}`);
+            return;
+        }
+
+        while (storeEvent.Next()) {
+            const matrixIds = storeEvent.MatrixId.split(";");
+
+            for (const room of rooms) {
+                const reactionEventId = await intent.underlyingClient.unstableApis.addReactionToEvent(
+                    room,
+                    matrixIds[0],
+                    reaction.emoji.id ? `:${reactionName}:` : reactionName
+                );
```

Comment:
We stumbled on an issue on an hosted bridge : if there are multiple Matrix rooms linked to a Discord channel, we can get the following message : `Relations must be in the same room`
If it fails on a room, then further rooms are ignored. Maybe a `try/catch` would be a nice addition to ignore this issue.

Reply:
I've rebased the branch with `develop` and resolved this.
Feel free to test it whenever you find time for it.